metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchQueryBuilder.java

```diff
@@ -89,41 +94,46 @@ private static Set<Pair<String, Float>> getStandardFields(@Nonnull EntitySpec en
 
       String fieldName = fieldSpec.getSearchableAnnotation().getFieldName();
       double boostScore = fieldSpec.getSearchableAnnotation().getBoostScore();
-      fields.add(Pair.of(fieldName, (float) (boostScore)));
+      fields.add(Pair.of(fieldName, (float) boostScore));
 
       FieldType fieldType = fieldSpec.getSearchableAnnotation().getFieldType();
       if (TYPES_WITH_DELIMITED_SUBFIELD.contains(fieldType)) {
-        fields.add(Pair.of(fieldName + ".delimited", (float) (boostScore * 0.4)));
+        fields.add(Pair.of(fieldName + ".delimited", ((float) boostScore) * 0.4f));
       }
       if (FieldType.URN_PARTIAL.equals(fieldType)) {
-        fields.add(Pair.of(fieldName + ".delimited", (float) (boostScore * 0.4)));
+        fields.add(Pair.of(fieldName + ".delimited", ((float) boostScore) * 0.4f));
       }
     }
 
     return fields;
   }
 
   private static Optional<QueryBuilder> getPrefixQuery(@Nonnull EntitySpec entitySpec, String query) {
-    BoolQueryBuilder finalQuery =  QueryBuilders.boolQuery();
+    BoolQueryBuilder finalQuery = QueryBuilders.boolQuery();
+    if (query.contains("\"")) {
+      finalQuery.should(QueryBuilders.termQuery("urn", query.replaceAll("\"", ""))
+              .boost(Float.parseFloat((String) PRIMARY_URN_SEARCH_PROPERTIES.get("boostScore")) * EXACT_MATCH_BOOST_FACTOR)
+              .queryName("urn"));
+    }
+
     entitySpec.getSearchableFieldSpecs().stream()
             .map(SearchableFieldSpec::getSearchableAnnotation)
             .filter(SearchableAnnotation::isQueryByDefault)
-            .filter(SearchableAnnotation::isEnableAutocomplete)
+            .filter(SearchableAnnotation::isEnableAutocomplete) // Proxy for identifying likely exact match fields
             .filter(e -> TYPES_WITH_DELIMITED_SUBFIELD.contains(e.getFieldType()))
-            .forEach(fieldSpec -> finalQuery.should(
-                    QueryBuilders.matchPhrasePrefixQuery(fieldSpec.getFieldName() + ".delimited", query)
-                            .boost((float) fieldSpec.getBoostScore())));
+            .forEach(fieldSpec -> {
+              finalQuery.should(QueryBuilders.matchPhrasePrefixQuery(fieldSpec.getFieldName() + ".delimited", query)
+                      .boost((float) fieldSpec.getBoostScore())
+                      .queryName(fieldSpec.getFieldName())); // less than exact
+              if (query.contains("\"")) {
+                finalQuery.should(QueryBuilders.termQuery(fieldSpec.getFieldName() + ".keyword", query.replaceAll("\"", ""))
+                        .boost(Float.parseFloat((String) PRIMARY_URN_SEARCH_PROPERTIES.get("boostScore")) * EXACT_MATCH_BOOST_FACTOR)
```

Comment:
So again expressing what I mentioned above, boosting cases where there are quotations in the query string I don't think will be sufficient for what users expect. They expect that searching for a name of a Table by exact string will find that table, with or without quotations

Reply:
I'll remove the quote conditional